File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r=0,len(heights)-1
        mxarea=0
        while l<r:
            h=min(heights[l],heights[r])
            w=r-l
            area=h*w
            mxarea=max(mxarea,area)
            if heights[l]<heights[r]:
                l+=1
            else:
                r-=1
        return mxarea
        

File: Data Structures & Algorithms/trapping-rain-water/submission-0.java
class Solution {
    public int trap(int[] height) {
        int n=height.length;
        int water=0;
        for(int i=0;i<n;i++){
            int lmax=0,rmax=0;
            for(int j=0;j<=i;j++) lmax=Math.max(lmax,height[j]);
            for(int j=i;j<n;j++) rmax=Math.max(rmax,height[j]);
            water+=Math.min(lmax,rmax)-height[i];

        }return water;
        
    }
}
